src/event-engine/states/receive_failed.ts

```diff
@@ -0,0 +1,45 @@
+import { State } from '../core/state';
+import { Cursor } from '../../models/Cursor';
+import { Effects } from '../effects';
+import { Events, reconnect, restore, subscriptionChange, unsubscribeAll } from '../events';
+import { PubNubError } from '../../core/components/endpoint';
+import { HandshakingState } from './handshaking';
+import { UnsubscribedState } from './unsubscribed';
+
+export type ReceiveFailedStateContext = {
+  channels: string[];
+  groups: string[];
+  cursor: Cursor;
+
+  reason: PubNubError;
+};
+
+export const ReceiveFailedState = new State<ReceiveFailedStateContext, Events, Effects>('RECEIVE_FAILED');
+
+ReceiveFailedState.on(reconnect.type, (context, event) =>
+  HandshakingState.with({
+    channels: context.channels,
+    groups: context.groups,
+    cursor: {
+      timetoken: event.payload?.timetoken ?? '0',
+      region: event.payload?.region ?? 0,
+    },
```

Comment:
Should we check the `context` for cursor information instead of defaulting to `0`?

Reply:
unified across the states